tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+
+        LOGGER.info("Step 2: Create soft failures and perform IAM user creation")
+        for pod in self.ctrl_pod_list:
+            LOGGER.info("Step 2.1: Create soft failure on pod %s", pod)
+            resp = self.node_master_list[0].kill_process_in_container(
+                pod_name=pod, container_name=self.container_name, pid=self.pid, safe_kill=True)
+            LOGGER.info("Response: %s", resp)
+            LOGGER.info("Step 2.1: Created soft failure")
+            LOGGER.info("Step 2.2: Create an IAM user")
+            user = self.ha_obj.create_iam_user_with_header(num_users=1, header=self.header)
+            LOGGER.debug("Created user: %s", user)
+            if pod == self.ctrl_pod_list[-1]:
+                assert_utils.assert_equal(user, None, "Expected failure in IAM user creation")
+                LOGGER.info("Step 2.2: Failed IAM user creation as expected")
+            else:
+                assert_utils.assert_not_equal(user, None, "Expected successful IAM user creation")
+                self.s3_clean.update(user)
+                self.uids.append(user[list(user.keys())[0]]["user_name"])
+                LOGGER.info("Step 2.2: Successfully created IAM user")
+            d_user = del_users.pop(list(del_users.keys())[0])
+            LOGGER.info("Step 2.3: Delete iam user %s", d_user["user_name"])
+            d_user_dict = {d_user["user_name"]: d_user}
+            resp = self.ha_obj.delete_iam_user_with_header(user=d_user_dict, header=self.header)
+            if pod == self.ctrl_pod_list[-1]:
+                assert_utils.assert_false(resp[0], "Expected failure in IAM user deletion")
+                LOGGER.info("Step 2.3: Failed IAM user deletion as expected")
+                self.uids.append(d_user["user_name"])
+                self.s3_clean.update(d_user_dict)
+            else:
+                assert_utils.assert_true(resp[0], "Expected successful IAM user deletion")
+                LOGGER.info("Step 2.3: Successfully deleted IAM user")
+
+            LOGGER.info("Step 2.4: Verify if %s IAM users are persistent across control pod "
+                        "soft failures", self.uids)
+            for user in self.uids:
+                resp = self.rest_iam_user.get_iam_user(user)
+                assert_utils.assert_equal(int(resp.status_code), HTTPStatus.OK.value,
+                                          f"Couldn't find user {user} after control pod "
+                                          "failover")
+                LOGGER.info("Step 2.4: User %s is persistent: %s", user, resp)
+        LOGGER.info("Step 2: Created soft failures and performed IAM user creation")
+
+        LOGGER.info("Step 3: Checking cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp[1])
+        LOGGER.info("Step 3: Cluster status should be bad as control pods are in failed state")
```

Comment:
So we are assuming that in step 2, we won't be reaching 60s delay and cluster will be still be failed?

Reply:
Yes, so once we create soft failure in last pod, cluster should be in failed state as restart will be delayed.